components/longestStreak/GameHUD.tsx

```diff
@@ -0,0 +1,85 @@
+import { useDispatch } from "react-redux";
+import {
+  checkNumberNotSelected,
+  calculatePlayerScore,
+} from "../../pages/api/games/longestStreak";
+import {
+  setStage,
+  STAGE,
+  reset,
+  initializeGame,
+} from "../../redux/longestStreakSlice";
+import { Button } from "../ui/Button";
+
+export default function GameHUD({ data, gameState, user }) {
```

Comment:
make sure you understand why I moved this code into a different file, and how it benefits the codebase by extracting this component out.

Reply:
For sure, I'll revise it further.